plugin/src/main/kotlin/org/gradle/dependencygraph/extractor/DependencyExtractor.kt

```diff
@@ -226,13 +230,12 @@ abstract class DependencyExtractor :
     }
 
     private fun coordinates(component: ResolvedComponentResult): DependencyCoordinates {
-        // TODO: Consider and handle null moduleVersion
-        val moduleVersionIdentifier = component.moduleVersion!!
-        return DependencyCoordinates(
-            moduleVersionIdentifier.group,
-            moduleVersionIdentifier.name,
-            moduleVersionIdentifier.version
-        )
+        val mv = component.moduleVersion
+        return if (mv != null) {
+            DependencyCoordinates(mv.group, mv.name, mv.version)
+        } else {
+            DependencyCoordinates("unknown", "unknown", "unknown")
+        }
```

Comment:
Suggestion:

```suggestion
        return component.moduleVersion?.let { DependencyCoordinates(it.group, it.name, it.version) } ?: 
            DependencyCoordinates("unknown", "unknown", "unknown")
```

or

```suggestion
        return component.moduleVersion?.run { DependencyCoordinates(group, name, version) } ?: 
            DependencyCoordinates("unknown", "unknown", "unknown")
```

Reply:
I had that originally, but find it harder to read.